ChangeLog.d/alignment-perf.txt

```diff
@@ -0,0 +1,4 @@
+Features
+   * Improvements to the code used to access unaligned and byte-swapped memory
+     reduce code size and improve performance (depending on compiler and target
+     architecture).
```

Comment:
This ChangeLog entry reads very strangely, possibly missing a comma between `memory` and `reduce`, but maybe `reduce` and `improve` should be `reducing` and `improving`. However, work started from issue 1666 "Speed up XORing of large numbers", but that is nothing to do with unaligned or byte-swapped memory. (Hence my feeling that this would have been better as multiple PRs, but we'll keep with just the one now).

As a concrete suggestion, how about

```
   * General performance improvements by accessing multiple bytes at a time.
     Fixes #1666.
   * Improvements to use of unaligned and byte-swapped memory, reducing code
     size and improving performance (depending on compiler and target
     architecture).
```

Reply:
Possible also

```
Changes
   * Mixed-endian systems are explicitly not supported any more.
```

Reply:
> Mixed-endian systems are explicitly not supported any more.

That's technically a breach of backward compatibility. Mixed-endian systems used to work. I think it's an acceptable breach because AFAIK, mixed-endian systems are vintage (and may well still be running in production in banks or power plants, but they aren't running new software like mbedtls).

This should be under “Requirements changes”, not the generic “Changes” (which is for non-functional differences).